.github/workflows/consumer_contract_tests.yml

```diff
@@ -0,0 +1,82 @@
+name: Consumer contract tests
+# The purpose of this workflow is to validate the service level contract
+# using the Pact framework.
+#
+# More details on Contract Testing can be found in our handbook
+#
+# https://broadworkbench.atlassian.net/wiki/spaces/IRT/pages/2660368406/Getting+Started+with+Pact+Contract+Testing
+#
+#
+#
+# NOTE: The publish-contracts workflow will use the latest commit of the branch that triggers this workflow to publish the unique consumer contract version to Pact Broker.
+
+on:
+  pull_request:
+    branches: [ '**' ]
```

Comment:
I'm not clear about this syntax-- does it mean that it ignores branches?

Reply:
iirc it means this will run on pull requests against all branches. I'm also curious about whether we need to clean up PR pacts

Reply:
Pacts are published [here](https://pact-broker.dsp-eng-tools.broadinstitute.org)

TBH, I'm not sure if we need to clean up PR pacts either.
They're indexed by commit hash, so once it's merged, it won't generate a separate pact.
It's also pretty compact in the grand scheme of things, and they can be removed manually; so I'm not especially concerned with it in the short term, since conventions are still being established.

It's plausible we should restrict it to PRs to `main`, though (not that it's a huge difference, since most of our PRs are against `main`).

Reply:
There is no need to clean up PR pacts on the Pact Broker. As @blakery said, normally we restrict PRs to trigger on `main`. When you push code to PR branch, the contract tests will run. Upon successful test run, this workflow will publish the contract to Pact Broker tagged with the latest commit, which is the version the Pact Broker cares about when calculating contract changes. Now the key part is, when your team decides everything looks good and merge PR to `main`, the same contract tests will run again which shouldn't cause no surprise, the contract will be published but tagged with the merge commit. Your repo HEAD is now tagged with merge commit, it's the same in Pact Broker. Furthermore, DevOps will be able to grab the latest commit for releases. Everything is in sync.